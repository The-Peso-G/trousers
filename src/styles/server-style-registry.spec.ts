/**
 * @jest-environment node
 */

import ServerStyleRegistry from './server-style-registry';

describe('ServerStyleRegistry', () => {
    let registry: ServerStyleRegistry;

    beforeEach(() => {
        registry = new ServerStyleRegistry();
    });

    it('registers a new style', () => {
        const key = '1';
        const style = 'background-color:red;';

        registry.register(key, style);

        expect(registry.get()).toMatchSnapshot();
    });

    it('registering an existing key does not update the style', () => {
        const key = '1';
        const style = 'background-color:red;';
        const styleBlue = 'background-color:blue;';

        registry.register(key, style);
        registry.register(key, styleBlue);

        expect(registry.get()).toMatchSnapshot();
    });

    it('detects an existing style', () => {
        const key = '1';
        const style = 'background-color:red;';

        registry.register(key, style);

        expect(registry.has('1')).toBe(true);
    });

    it('detects a non existing style', () => {
        expect(registry.has('123')).toBe(false);
    });

    it('renders registered styles in the right format', () => {
        registry.register('1', 'background-color:red;');
        registry.register('2', 'background-color:blue;');

        expect(registry.get()).toMatchSnapshot();
    });
});
