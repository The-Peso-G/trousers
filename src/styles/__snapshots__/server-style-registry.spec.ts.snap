// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerStyleRegistry registering an existing key does not update the style 1`] = `
<style data-trousers={1}>
  1{background-color:red;}
</style>
`;

exports[`ServerStyleRegistry registers a new style 1`] = `
<style data-trousers={1}>
  1{background-color:red;}
</style>
`;

exports[`ServerStyleRegistry renders registered styles in the right format 1`] = `
<style data-trousers={1,2}>
  1{background-color:red;}
  2{background-color:blue;}
</style>
`;
